File: FFinder/app/src/main/java/com/ffinder/android/enums/SearchStatus.java
package com.ffinder.android.enums;

import android.content.Context;
import com.ffinder.android.R;

/**
 * Created by SiongLeng on 1/9/2016.
 */
public enum SearchStatus {
    Starting,
    CheckingData,
    WaitingUserRespond,
    WaitingUserLocation,
    Geocoding,
    End;

    public String getMessage(Context context){
        switch (this){
            case Starting:
                return context.getString(R.string.search_status_starting_msg);
            case CheckingData:
                return context.getString(R.string.search_status_checking_data_msg);
            case WaitingUserRespond:
                return context.getString(R.string.search_status_waiting_user_respond_msg);
            case WaitingUserLocation:
                return context.getString(R.string.search_status_waiting_user_location_msg);
            case Geocoding:
                return context.getString(R.string.search_status_geodecoding_msg);
            case End:
                return null;
        }

        return this.name();
    }


}

File: FFfinder_mac/FFfinder/FFfinder/SearchStatus.swift
//
//  SearchStatus.swift
//  FFfinder
//
//  Created by Siong Leng Ho on 16/10/16.
//  Copyright © 2016 Lightweight Studio Apps. All rights reserved.
//

import Foundation
import EVReflection
enum SearchStatus:String, EVRawString{
    
    case Starting = "Starting"
    case CheckingData = "CheckingData"
    case WaitingUserRespond = "WaitingUserRespond"
    case WaitingUserLocation = "WaitingUserLocation"
    case End = "End"
    
    public static func getMessage(_ status:SearchStatus) -> String{
        switch status {
            case .Starting:
                return "search_status_starting_msg".localized
            case .CheckingData:
                return "search_status_checking_data_msg".localized
            case .WaitingUserRespond:
                return "search_status_waiting_user_respond_msg".localized
            case .WaitingUserLocation:
                return "search_status_waiting_user_location_msg".localized
            default:
                return ""
        }
    }
    
    
}
